scripts/lib.sh

```diff
@@ -374,62 +330,36 @@ get_extensions_from_dynatrace() {
   rm -rf "${CLUSTER_EXTENSIONS_TMPDIR}"
 }
 
-upload_correct_extension_to_dynatrace() {
-  SERVICES_FROM_ACTIVATION_CONFIG_STR=$1
-
-  cd "${EXTENSIONS_TMPDIR}" || exit
+get_and_install_extensions() {
+  debug "Getting and Installing Dynatrace GCP Extensions List"
 
-  AMOUNT_OF_EXTENSIONS_TO_UPLOAD=0
-  AMOUNT_OF_NOT_UPLOADED_EXTENSIONS=0
-  AMOUNT_OF_NOT_ACTIVATED_EXTENSIONS=0
+  info ""
+  info "- getting list of extensions"
+  LIST_OF_EXTENSIONS=$(dt_api "/api/v2/extensions")
 
-  for EXTENSION_ZIP in *.zip; do
-    EXTENSION_FILE_NAME="$(basename "$EXTENSION_ZIP" .zip)"
+  NUMBER_OF_EXTENISIONS_ON_TENANT=$(echo "$LIST_OF_EXTENSIONS" | jq -r '.totalCount')
+  
+  if [ "$NUMBER_OF_EXTENISIONS_ON_TENANT" -gt 0 ]; then
+      info ""
+      info "- there are extensions alredy on the tenant:"
+  else
+      info "- searching for available extensions..."
 
-    unzip -j -q "$EXTENSION_ZIP" "extension.zip"
-    unzip -p -q "extension.zip" "extension.yaml" >"$EXTENSION_FILE_NAME".yaml
+      AVAILABLE_EXTENSIONS=$(dt_api "/api/v2/hub/items?itemType=EXTENSION2")
```

Comment:
Is there no paging? If there is 5k of extensions it will still work? Is this possible to have this contains filter on the source?

Reply:
> Is there no paging? If there is 5k of extensions it will still work? Is this possible to have this contains filter on the source?

@mmajcher sure, working on this

Reply:
@mmajcher there is limitations on the backend side while using pagination and specifying extensions type... I will rewrite it in different way.